ACD-618 (#30)
Actindo-AG/Documentation: Channels/Integration/01c_Salesforce.md

# Manage the Salesforce Commerce Cloud connection

> To establish a connection to a Salesforce Commerce Cloud shop, there are several particularities to consider. Therefore, the creation and the configuration of the Salesforce Commerce Cloud connection are described in detail below.

Author: ```suggestion
To establish a connection to a Salesforce Commerce Cloud shop, there are several particularities to consider. Therefore, the creation and the configuration of the Salesforce Commerce Cloud (Salesforce) connection are described in detail below.
```

## Configuration of Salesforce Commerce Cloud

> Before a connection between Salesforce Commerce Cloud and *Actindo Core1 Platform* can be established, there are two configuration processes that must be previously performed in Salesforce Commerce Cloud. 

Author: ```suggestion
Before a Salesforce Commerce Cloud connection can be established, there are two configuration processes that must be performed. 
```

### Create jobs

> In Salesforce Commerce Cloud, data are uploaded via XML files, which are then imported with a job into the shop. There are two jobs, one for data import and one for inventory import. 
> 
> Please note that these two jobs must be configured in the Salesforce Commerce Cloud shop in the exact order and using the specified values as described below.

Author: ```suggestion
In Salesforce, data are uploaded via XML files, which are then imported with a job into the shop. There are two jobs, one for data import and one for inventory import.    
Please note that these two jobs must be configured in your Salesforce account in the exact order and using the specified values as described below.
```

#### Prerequisites

- A Salesforce Commerce Cloud account has been created.
- You have the appropriate user rights.

#### Procedure

> 1. In a browser window, go to the *Salesforce Commerce Cloud* platform and log in to your account.

Author: ```suggestion
1. In a new browser window, go to the *Salesforce Commerce Cloud* platform and log in to your account.
```

> 2. Click the *Select a Site* drop-down list and select the applicable shop.  

Author: ```suggestion
2. Click the *Select a Site* drop-down list in the upper left corner of the header and select the applicable shop. All available shops are displayed in the list.  
```

> 2. Click the *Select a Site* drop-down list and select the applicable shop.  
>     The site business manager page is displayed.

Author: ```suggestion
2. Click the *Select a Site* drop-down list in the upper left corner of the header and select the applicable shop. All available shops are displayed in the list.  
    The *Welcome to the Business Manager* page is displayed.
```

    ![Welcome to Business Manager](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/BusinessManager.png "[Welcome to Business Manager]")

> 3. Using the site navigation bar at the top of the page, go to *Administration > Operations > Jobs*.  

Author: ```suggestion
3. Click the *Administration* drop-down list in the navigation bar at the top of the page. 
   The *Administration page is displayed.
   
4. Click first the *Operations* and then the *Jobs* module.  
```

    The *Jobs* page is displayed.

    ![Jobs](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Jobs.png "[Jobs]")

4. Click the [New Job] button in the upper right corner to create the inventory import job.  

>     The *New Job* window is displayed on the right of the workspace.

Author: ```suggestion
    The *New Job* window is displayed on the right of the workspace.
```
window oder page? Fehlt hier ein *side* ach dem right? Woher kommt die Info?

    ![New Job](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Jobs.png "[New Job]")

> 5. Enter the job name "MSN-InventoryImport" in the *ID* field and, if desired, a job description in the *Description* field, and click the [Create] button.  
>     The new job has been created. The job page is displayed.

Author: ```suggestion
5. Enter a name for the inventory import job, for example **InventoryImport**, in the *ID* field and, if desired, a job description in the *Description* field, and click the [Create] button.  
    The new job has been created. The *Jobs* page is displayed.
```
Das MNS ist Nintendo spezifisch und sollte hier nicht auftauchen (My Nintendo Store = MNS)

> 6. In the navigation bar at the top of the page, click the *Job Steps* tab and select *Configure a step* to create a new step. 
>     The *Select and Configure Step* side bar is displayed on the right of the workspace.

Author: ```suggestion
6. Click the *Job Steps* tab in the header of the *"JobID"* page and click the [](Add) button.
   A new flow with a step has been added to the job. The *Select and Configure Step* side bar is displayed on the right side of the page.
```
Müssen wir noch prüfen, ob das so beim Anlegen eines neuen Flows/steps funktioniert. Aber auf jeden fall kann ich nirgends drauf klicken, bevor ich keinen Flow angelegt habe. Vgl. auch https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2FDWAPI%2Fxsd%2FSchemas.html

7. In the *Select and Configure Step* side bar, the following values must be configured: 

>     | Field                    | Value     |
>     |--------------------------|-------------------------|
>     | Context                  | Organization            |
>     | ID                       | ImportInventory lists   |
>     | Description              | -                       |
>     | WorkingFolder            | import/inventory        |   
>     | FileNamePattern          | ac_.*_inventory\\.xml   | 
>     | NofilesFoundHandling*    | NO_FILES_FOUND          |
>     | ImportMode*              | Merge                   |  
>     | ImportFailedHandling*    | WARN                    |  
>     | AfterImportFileHandling* | Archive                 |
>     | ArchiveFolder            | archive/inventory       |  
> 
> [comment]: <> (Scope: Organization - Wo einzustellen? Relevant hier? Job FLow -> Was ist? Muss man ein Job flow erstellen, bevor man ein Step erstellt?)

Author: ```suggestion
    | Field                    | Value     |
    |--------------------------|-------------------------|
    | WorkingFolder            | import/inventory        |   
    | FileNamePattern          | ac_.*_inventory\.xml   | 
    | NoFilesFoundHandling    | NO_FILES_FOUND          |
    | ImportMode              | Merge                   |  
    | ImportFailedHandling    | WARN                    |  

 > [Info] The values in the *AfterImportFileHandling* and the *ArchiveFolder* fields can berbitrarily selected.
```

> 8. Click on [Assign].  

Author: ```suggestion
8. Click the [Assign] button in the bottom right corner.  
```

    The job step has been configured.

> 

Author: Würde getrennte Procedures für die beiden Jobs machen, dann sind es auch nicht so viele steps.

9. Click again the [New Job] button in the upper right corner to create the data import job.  
    The *New Job* window is displayed on the right side of the workspace.

10. Enter the job name "MSN-SiteDataImport" in the *ID* field and, if desired, a job description in the *Description* field, and click the [Create] button.  

>     The new job has been created. The job page is displayed.

Author: anpassen wie oben

> 9. Click again the [New Job] button in the upper right corner to create the data import job.  
>     The *New Job* window is displayed on the right side of the workspace.
> 
> 10. Enter the job name "MSN-SiteDataImport" in the *ID* field and, if desired, a job description in the *Description* field, and click the [Create] button.  
>     The new job has been created. The job page is displayed.

Author: ```suggestion
9. Click the [New Job] button in the upper right corner to create the data import job. 
    The *New Job* window is displayed on the right side of the page.

10. Enter a name for the data import job, for example **SiteDataImport**, in the *ID* field and, if desired, a job description in the *Description* field, and click the [Create] button.  
  The new job has been created. The *"JobID"* page is displayed.
  
  > [Info] Note this ID, as it must be specified in the *Omni-Channel* module when establishing the connection to Salesforce.
  
11. Click the *Job Steps* tab in the header of the *"JobID"* page and click the [] (Add) button.
   A new flow has been added to the job. 
   
12. Select the **Organization** option as a scope for the flow.
```
add button hinzufügen

>     The MSN-SiteDataImport imports different types of data and is therefore a multi-step job. Repeat steps **6** to **8** to create and configure the following
>     job steps:

Author: ```suggestion
11. Click the [Configure a step] button in the flow box and select the *StandardProductsImport* step.
   The *Select and Configure Step* side bar is displayed on the right side of the page.
   
12. Configure the following values in the corresponding fields of the *Select and Configure Step* side bar: 
 ```

    - StandardProductsImport (Scope: Organization)  

        | Field                    | Value     |
        |--------------------------|-------------------------|
        | Context                  | Organization            |
        | ID                       | StandardProductsImport  |
        | Description              | -                       |
        | WorkingFolder            | import/catalogs         |   
        | FileNamePattern          | ac_.\*_catalog\_.*\\.xml| 
        | NofilesFoundHandling*    | NO_FILES_FOUND          |
        | ImportMode*              | Merge                   |  
        | ImportFailedHandling*    | WARN                    |  
        | AfterImportFileHandling* | Archive                 |
        | ArchiveFolder            | archive/catalogs        |  

>         > [Info] The Import/catalogs working folder must be previously created in WebDAV. Otherwise, the data cannot be imported. For detailed information, see [Using WebDAV](https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2Fcontent%2Fb2c_commerce%2Ftopics%2Fimport_export%2Fb2c_using_web_dav.html).

Author: ```suggestion
```

>     - BundleProductsImport (Scope: Organization)  

Author: 13. Click the [] (Add) button below the first flow.
   A new flow has been added below the first flow. 
   
14. Select the **Organization** option as a scope for the flow.

15. Click the [Configure a step] button in the flow box and select the *BundleProductsImport* step.
   The *Select and Configure Step* side bar is displayed on the right side of the page.
   
16. Configure the following values in the corresponding fields of the *Select and Configure Step* side bar for the *BundleProductsImport* step:

>     - BundleProductsImport (Scope: Organization)  
> 
>         | Field                    | Value     |
>         |--------------------------|-------------------------|
>         | Context                  | Organization            |
>         | ID                       | BundleProductsImport    |
>         | Description              | -                       |
>         | WorkingFolder            | import/catalogs         |   
>         | FileNamePattern          | ac_.\*_bundles\_.*\\.xml| 
>         | NofilesFoundHandling*    | NO_FILES_FOUND          |
>         | ImportMode*              | Merge                   |  
>         | ImportFailedHandling*    | WARN                    |  
>         | AfterImportFileHandling* | Archive                 |
>         | ArchiveFolder            | archive/catalogs        | 

Author: ```suggestion
13. Click the [] (Add) button below the first flow.
   A new flow has been added below the first flow. 
   
14. Select the **Organization** option as a scope for the flow.

15. Click the [Configure a step] button in the flow box and select the *BundleProductsImport* step.
   The *Select and Configure Step* side bar is displayed on the right side of the page.
   
16. Configure the following values in the corresponding fields of the *Select and Configure Step* side bar for the *BundleProductsImport* step: 

        | Field                    | Value     |
        |--------------------------|-------------------------|
        | WorkingFolder            | import/catalogs         |   
        | FileNamePattern          | ac_.\*_bundles\_.*\\.xml| 
        | NoFilesFoundHandling*    | NO_FILES_FOUND          |
        | ImportMode*              | Merge                   |  
        | ImportFailedHandling*    | WARN                    |  

   > [Info] The values in the *AfterImportFileHandling* and the *ArchiveFolder* fields can be arbitrarily selected.
```
Add button hinzufügen

>     - CategoryAssignmentImport (Scope: Organization)  
> 
> 
>         | Field                    | Value     |
>         |--------------------------|-------------------------|
>         | Context                  | Organization            |
>         | ID                       | CategoryAssignmentsImport|
>         | Description              | -                       |
>         | WorkingFolder            | import/catalogs         |   
>         | FileNamePattern          | ac_.\*_assignments\_.*\\.xml| 
>         | NofilesFoundHandling*    | NO_FILES_FOUND          |
>         | ImportMode*              | Merge                   |  
>         | ImportFailedHandling*    | WARN                    |  
>         | AfterImportFileHandling* | Archive                 |
>         | ArchiveFolder            | archive/catalogs        |  

Author: ```suggestion
 13. Click the [] (Add) button below the secondflow.
   A new flow has been added below the second flow. 
   
14. Select the **Organization** option as a scope for the flow.

15. Click the [Configure a step] button in the flow box and select the *CategoryAssignmentImport* step.
   The *Select and Configure Step* side bar is displayed on the right side of the page.
   
16. Configure the following values in the corresponding fields of the *Select and Configure Step* side bar for the *CategoryAssignmentImport* step: 

        | Field                    | Value     |
        |--------------------------|-------------------------|
        | WorkingFolder            | import/catalogs         |   
        | FileNamePattern          | ac_.\*_assignments\_.*\\.xml| 
        | NoFilesFoundHandling    | NO_FILES_FOUND          |
        | ImportMode              | Merge                   |  
        | ImportFailedHandling    | WARN                    |  

   > [Info] The values in the *AfterImportFileHandling* and the *ArchiveFolder* fields can be arbitrarily selected.
```
add button hinzufügen

    - PricebookImport/ImportPriceBook (Scope: Organization)

        | Field                    | Value     |
        |--------------------------|-------------------------|
        | Context                  | Organization            |
        | ID                       | PricebookImport|
        | Description              | -                       |
        | WorkingFolder            | import/pricebooks       |   
        | FileNamePattern          | ac_.\*\_price_book\\.xml| 
        | NofilesFoundHandling*    | NO_FILES_FOUND          |
        | ImportMode*              | Merge                   |  
        | ImportFailedHandling*    | WARN                    |  
        | AfterImportFileHandling* | Archive                 |
        | ArchiveFolder            | archive/pricebooks      |  

>     > [Info] The Import/pricebooks working folder must be previously created in WebDAV. Otherwise, the data cannot be imported. For detailed information, see [Using WebDAV](https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2Fcontent%2Fb2c_commerce%2Ftopics%2Fimport_export%2Fb2c_using_web_dav.html).

Author: ```suggestion
```

>     > [Info] The Import/pricebooks working folder must be previously created in WebDAV. Otherwise, the data cannot be imported. For detailed information, see [Using WebDAV](https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2Fcontent%2Fb2c_commerce%2Ftopics%2Fimport_export%2Fb2c_using_web_dav.html).
> 
> 
>     - SearchReindex (Scope: client site, therefore the step configuration is not relevant for the *Actindo Core1 Platform*)
> 
> [comment]: <> (Evtl. dieser Step rauslassen? Oder etwas dazu erklären?)

Author: ```suggestion
17. Click the [Assign] button in the bottom right corner.    
    The step has been configured.
```

### Configure the Open Commerce API Settings

> Once the necessary jobs have been created, the API settings must be configured in the Salesforce Commerce Cloud shop.
> 
> There are two APIs in Salesforce Commerce Cloud: 
> - The shop API, which is used for the order data in the *Actindo Core1 Platform*
> - The data API, which is used for product data in the *Actindo Core1 Platform*
> 
> A Client ID must be generated in Salesforce Commerce Cloud and entered in the *client_id* field.
> 
> [comment]: <> (Wo/wie? Relevant zu erklären? Mehr Info benötigt! API Codes von Julian? Gehört das Code in der Doku oder kann der Kunde aus irgendwo rauskopieren?)

Author: ```suggestion
In Salesforce, there are two Open Commerce APIs: 
- The shop API, which is used for the order data in the *Actindo Core1 Platform*
- The data API, which is used for product data in the *Actindo Core1 Platform*

The corresponding API settings must be configured in the Salesforce as described below.
```

#### Prerequisites

> - The necessary jobs have been created in Salesforce Commerce Cloud.

Author: ```suggestion
```
muss das ne Voraussetzung sein? woher kommt die Info?

> - The Client ID has been generated.

Author: ```suggestion
The client ID has been generated in Salesforce.
```

> [comment]: <> (Stimmt das so? Wie/wo ist client ID erzeugt?)

Author: ```suggestion
```

#### Procedure

*Salesforce Commerce Cloud > Administration > Site Development > Open Commerce API Settings*

[comment]: <> (Screenshot von Open Commerce API Settings? Vgl. Procedure oben.)

1. Click the *Select type* drop-down list and select the appropriate option. The following options are available:
    - **Shop**  
        Select this option to configure the shop API.
    - **Data**  
        Select this option to configure the data API.

2. If desired, click the *Select Context* drop-down list and select the appropriate option. The following options are available:  
    - **Global (organization-wide)**  
        Select this option for the configuration to affect all sites in the organization.
    - Site-specific  
        Select the name of a site for the configuration to affect only that site. 

3. In the text field, edit the JSON document as necessary.

4. Click [Save].

[comment]: <> (Client permissions? To use OCAPI, you must first configure client permissions. These permissions control read and write access to specified resources. By default, no permission is granted... S. Salesfore Docu OCAPI Settings?)

## Create a Salesforce Commerce Cloud connection

> Create the connection to a Salesforce Commerce Cloud shop using the Salesforce Commerce Cloud driver. Further settings can only be configured after the connection has been established.

Author: ```suggestion
Create the connection to a Salesforce shop using the Salesforce Commerce Cloud driver. Further settings can only be configured after the connection has been established.
```

#### Prerequisites

> - Salesforce Commerce Cloud has been configured, see [Configuration of Salesforce Commerce Cloud](#configuration-of-salesforce-commerce-cloud). 

Author: ```suggestion
- The Salesforce Commerce Cloud has been configured, see [Configuration of Salesforce Commerce Cloud](#configuration-of-salesforce-commerce-cloud). 
```

- The *Salesforce Commerce Cloud* plugin has been installed.  

> [Info] For the *Omni-Channel* module version 4.1.0 or higher, the *Salesforce Commerce Cloud* plugin is required in at least version 1.2.0.

[comment]: <> (check!)

#### Procedure

*Omni-Channel > Settings > Tab CONNECTIONS*

> ![SSFC connection](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Connection.png "[SSFC connection]")

Author: ```suggestion
![SFCC connection](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Connection.png "[SFCC connection]")
```

1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button in the bottom right corner.    
  The *Create connection* view is displayed.

    ![Create connection](../../Assets/Screenshots/Channels/Settings/Connections/CreateConnection.png "[Create connection]")

2. Enter a name for the connection in the *Name* field.

3. Click the *Driver* drop-down list and select the *Salesforce Commerce Cloud* driver.  
  The *Credentials* section is displayed below the drop-down list.

>   > [Info] Salesforce Commerce Cloud offers the possibility to work with two separate environments, a staging one for customizing and testing purposes and a live one for production. Product data are uploaded to the staging environment and, if applicable, replicated in the live environment after acceptance. This structure is supported by the *Actindo Core1 Platform*. 

Author: ```suggestion
  > [Info] Salesforce offers the possibility to work with two separate environments, a staging environment for customizing and testing purposes and a live environment for production. The connection to the environments must be specified separately.
```

> 4. For a staging environment, the following credentials must be filled:

Author: ```suggestion
4. For the staging environment, enter the following credentials:
```

> 4. For a staging environment, the following credentials must be filled:
>     - Enter the shop URL in the *URL* field.  
>     - Enter the client ID in the *Client ID* field and the client password in the *Client password* field for the Salesforce Commerce Cloud shop API.
>     - Enter the user's business manager login details, that is, the username (email), user password and  the user API access token in the *Username*, *User password* and *Access token* fields. 

Author: ```suggestion
4. For the staging environment, enter the following credentials:
    - Enter the shop URL of the staging environment in the *URL* field.  
    - Enter the client ID in the *Client ID* field and the client password in the *Client password* field.
    - Enter the username (email address) in the *Username* field, the user password in the *User password* field and the user API access token in the *Access token* field. These so-called *business manager login data* are needed to work with orders.
```

> 5. If you want to work on a live environment as well, enable the *Production mode* toggle. In this case, the following credentials must be additionally filled:  

Author: ```suggestion
5. For the live environment, enter the following credentials and enable the *Production mode* toggle:
```

>     - *URL (Production)*
>     - *Client ID (Production)*
>     - *Client password (Production)*
>     - *Username (Production)*
>     - *User password (Production)*
>     - *Access token (Production)*

Author: ```suggestion
    - Enter the shop URL of the production environment in the *URL (Production)* field.  
    - Enter the corresponding client ID in the *Client ID (Production)* field and the client password in the *Client password (Production)* field.
    - Enter the username (email address) in the *Username (Production)* field, the user password in the *User password (Production)* field and the user API access token in the *Access token (Production)* field. These so-called *business manager login data* are needed to work with orders.
```

>     > [Info] As soon as a the production mode is enabled, the order and the inventory data are uploaded in production, whereas the product data are uploaded in staging.

Author: ```suggestion
    > [Info] As soon as a the production mode is enabled, the order import and the stock upload are synchronized with the production environment, whereas the product data are still synchronized with the staging environment.
```

6. Click the [SAVE] button.  

>     The connection has been created.
>   
>   The *CONNECTIONS* tab in the *Settings* menu entry of the *Omni-Channel* module is displayed when the connection has been established. The *Salesforce Commerce Cloud* connection is displayed in the list of connections.

Author: ```suggestion
  The connection has been created.   
  The *CONNECTIONS* tab in the *Settings* menu entry of the *Omni-Channel* module is displayed when the connection has been established. The *Salesforce Commerce Cloud* connection is displayed in the list of connections.
```

  ![Salesforce Commerce Cloud connection](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Connection.png "[Salesforce Commerce Cloud connection]")

12. If necessary, continue to [Configure the Salesforce Commerce Cloud connection](#configure-the-salesforce-commerce-cloud-connection).

[comment]: <> (Was genau passiert nach SAVE in Core1? Einige Minuten Zeit/Sync/Platform initialized? Satz aus Shopify genommen.)

## Configure the Salesforce Commerce Cloud connection   

After the connection to a Salesforce Commerce Cloud shop has been established, further settings can be configured for the connection.

#### Prerequisites

A Salesforce Commerce Cloud connection has been established.

#### Procedure

*Omni-Channel > Settings > Tab CONNECTIONS*

> ![SSFC connection](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Connection.png "[SSFC connection]")

Author: ```suggestion
![SFCC connection](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Connection.png "[SFCC connection]")
```

1. Click the Salesforce Commerce Cloud connection in the list of connections.   

>   The *Edit connection* view is displayed. By default, the *Credentials* tab is displayed.

Author: ```suggestion
  The *Edit connection* view is displayed. By default, the *Credentials* tab is selected.
```

    ![Edit connection credentials](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Credentials.png "[Edit connection credentials]")

2. Click the *Settings* tab.   

>   The *Settings* tab is displayed. By default, the *Price books* setting is displayed. 

Author: ```suggestion
  The *Settings* tab is displayed. By default, the *Price books* setting is selected. 
```

    ![Price books](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_PriceBooks.png "[Price books]")

> 3. Enter the price book ID in the *ID* field and the applicable currency ISO 4217 (three letter) code in the *Currency* field. Further price books can be added if necessary using the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.  

Author: ```suggestion
3. Enter the price book ID in the *ID* field and the applicable three letter currency code according to ISO 4217 in the *Currency* field for each price book in Salesforce. To add another price book, click the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.  
```

    > [Info] The price books must be set up manually, as they cannot be retrieved via API from Salesforce Commerce Cloud.

4.  Click the *Payment methods* menu entry in the left side bar.  

>     The payment methods setting is displayed in the right side bar.  

Author: ```suggestion
    The payment method settings are displayed on the right side.  
```

     ![Payment methods](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_PaymentMethods.png "[Payment methods]")

5. Enter the payment method ID in the *ID* field and a  description in the *Name* field. Further payment methods can be added if necessary using the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.  

6. Click the *Upload mode* menu entry in the left side bar.  

>     The upload setting is displayed in the right side bar.

Author: ```suggestion
    The upload mode settings are displayed on the right side.
```

    ![Upload mode](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_UploadMode.png "[Upload mode]")

> 7. Click the *Upload mode* drop-down list and select the appropriate option to define the system upload behavior when changes on product data have been made. The following options are available:  

Author: ```suggestion
7. Click the *Upload mode* drop-down list and select the appropriate mode to define the system upload behavior when changes on product data have been made. The following options are available:  
```

>     - *Immediately*  
>         Select this option to upload any changes as soon as they are made.
>     - *Delayed*  
>         Select this option to upload any changes after a specified period of time. When selected, the *Delay in minutes* field is displayed. By default, a 10 minute delay is predefined. This value can be changed if desired. 
>     - *Cron*  
>         Select this option to schedule the upload and perform it periodically at a fixed time, date or interval. When selected, the *Cron line* field is displayed to define the cron job. 

Author: ```suggestion
    - **Immediately**    
        Select this option to upload any changes as soon as they are made.
    - **Delayed**    
        Select this option to upload any changes after a specified period of time. When selected, the *Delay in minutes* field is displayed. By default, a 10 minute delay is predefined. This value can be changed if desired. 
    - **Cron**    
        Select this option to schedule the upload and perform it periodically at a fixed time, date or interval. When selected, the *Cron line* field is displayed to define the cron job. 
```

>       Alternatively, you can enable the *Apply from default* toggle to apply the predefined values.

Author: Welcher ist denn der predefined value? das sollten wir ergänzen, wenn wir den Satz übernehmen.

8. Click the *Tax classes* menu entry in the left side bar.  

>     The tax classes setting is displayed in the right side bar.

Author: ```suggestion
    The tax classes settings are displayed on the right side.
```

    ![Tax classes](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_TaxClasses.png "[Tax classes]")

> 9. Enter the applicable Salesforce Commerce Cloud tax code in the *Commerce Cloud Code* field and the corresponding tax code in the *Actindo Core1 Platform* in the *Actindo ID* field. The Actindo tax codes can be found in the *Taxes* module. Further tax codes can be added if necessary using the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.  

Author: ```suggestion
9. Enter the Salesforce Commerce Cloud ID for the tax class in the *Commerce Cloud Code* field and the corresponding Actindo tax class ID in the *Actindo ID* field. The Actindo tax class IDs can be found in the *Taxes* module. 

> [Info] Repeat the step **9** for each tax class in Salesforce. Further tax classes can be added by clicking the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.  
```
Problem: ich finde die Actindo Tax Class IDs nicht im Taxes Modul. Nachfragen, wo die angezeigt werden und genauer spezifizieren!

10. Click the *Ignored Product Attributes* menu entry in the left side bar.  

>     The ignored product attributes setting is displayed in the right side bar.

Author: ```suggestion
    The ignored product attributes settings are displayed on the right side.
```

    ![Ignored product attributes](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_IgnoredProductAttributes.png "[Ignored product attributes]")

> 11. Enter the ID of any fields that should not be imported in the *SFCC Attribute ID* field. The field IDs can be found in the catalog.xsd schema, see [catalog.xsd](https://documentation.b2c.commercecloud.salesforce.com/DOC3/index.jsp?topic=%2Fcom.demandware.dochelp%2FDWAPI%2Fxsd%2FSchemas.html). Further fields can be added if necessary using the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.  

Author: ```suggestion
11. Enter the ID of any attributes that should not be imported to Salesforce in the *SFCC Attribute ID* field. The corresponding field IDs are described in the *catalog.xsd* schema, see [catalog.xsd](https://documentation.b2c.commercecloud.salesforce.com/DOC3/index.jsp?topic=%2Fcom.demandware.dochelp%2FDWAPI%2Fxsd%2FSchemas.html). 
> [Info] Repeat the step **11** for each attribute to be ignored. Further attribute fields can be added by clicking the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.
```

>     > [Info] If there are custom fields that should not be imported, this must be specified here as well. Bear in mind that custom fields in *Salesforce Commerce Cloud* contain the prefix *c_", that is, *c_FieldName*.

Author: ```suggestion
    > [Info] If there are custom attributes that should not be imported to Salesforce, bear in mind to add the prefix **c_**, that is, **c_AttributeName**.
```

12. Click the *Default language* menu entry in the left side bar.  

>     The default language setting is displayed in the right side bar.

Author: ```suggestion
    The default language setting is displayed on the right side.
```

    ![Default language](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_DefaultLanguage.png "[Default language]")

> 13. Click the *Language used for default value* drop-down list and select the appropriate option. All available languages are displayed in the list.
> 
>     > [Info] In contrast to Salesforce Commerce Cloud, the *Actindo Core1 Platform* does not have a default language field. For this reason, a language must be specified to be used as default value.   

Author: ```suggestion
13. Click the *Language used for default value* drop-down list and select the language that should be used if the default language is selected in Salesforce . All available languages are displayed in the list.

    > [Info] In contrast to Salesforce Commerce Cloud, the *Actindo Core1 Platform* does not have a default language field. For this reason, a language must be specified to be used as default value.   
```

14. Click the *Skip as variation attribute* menu entry in the left side bar.  

>     The skip as variation attribute setting is displayed in the right side bar.

Author: ```suggestion
    The skip as variation attribute settings are displayed on the right side.
```

    ![Skip as variation attribute](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_SkipVariationAttributes.png "[Skip as variation attribute]")

> 15. Enter the ID of any variant fields that should not be imported in the *SFCC Attribute ID* field. Further fields can be added if necessary using the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.  
> 
>     > [Info] In contrast to the *Actindo Core1 Platform*, variant attributes are not necessarily unique. To avoid import problems, it is recommended not to import these fields. 

Author: ```suggestion
15. Enter the Salesforce ID of the variant that should not be imported to Salesforce in the *SFCC Attribute ID* field.   

> [Info] Repeat the step **15** for each variant to be ignored. Further variant fields can be added by clicking the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button.
```

> 16. In the current version, the image upload is not supported, and therefore the *View types* menu entry is not relevant for the settings. 

Author: ```suggestion
```
step nummern im Folgenden anpassen

17. Click the *Job ID data import* menu entry in the left side bar.  
    The Job ID data import setting is displayed in the right side bar.   

    ![Job ID data import](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_JobIDDataImport.png "[Job ID data import]")

18. Enter the applicable Salesforce Commerce Cloud job ID in the *Job ID data import* field. 

19. Click the *Job ID inventory import* menu entry in the left side bar.  
    The Job ID inventory import setting is displayed in the right side bar.   

    ![Job ID inventory import](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_JobIDInventoryImport.png "[Job ID inventory import]")

20. Enter the applicable Salesforce Commerce Cloud job ID in the *Job ID inventory import* field.

21. Click the *Order status fields* menu entry in the left side bar.  
    The order status fields setting is displayed in the right side bar.

>     ![Order status fields](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_OrderStatusFields.png "[Order status fields]")

Author: Überlegen, ob wir das hier beschreiben wollen (mit Verweis auf Create jobs) oder im Create jobs Abschnitt selbst und hier nur den Hinweis hinterlassen, dass es bereits konfiguriert wurde. Wo ist das sinnvoller?

> 21. Click the *Order status fields* menu entry in the left side bar.  
>     The order status fields setting is displayed in the right side bar.
> 
>     ![Order status fields](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_OrderStatusFields.png "[Order status fields]")
> 
> 22. Enter the IDs of the fields to be used for order status, payment status and deliver status in the corresponding fields. Alternatively, you can enable the *Apply from default* toggle to apply the predefined values.

Author: ```suggestion
21. Click the *Order status fields* menu entry in the left side bar.  
    The order status fields settings are displayed on the right side.

    ![Order status fields](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_OrderStatusFields.png "[Order status fields]")

22. Enter the custom IDs of the corresponding status fields in the *Order status*, *Payment status* and *Deliver status* fields. Alternatively, you can enable the *Apply from default* toggle to apply the Salesforce default values.

> [Info] Bear in mind to add the prefix **c_** for any custom field names, that is, **c_StatusID**.
```

> 23. Click the *Upload stock only* menu entry in the left side bar.  
>     The upload stock only setting is displayed in the right side bar.
> 
>     ![Upload stock only](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_UploadStockOnly.png "[Upload stock only]")
> 
> 24. Select the *Upload stock only* checkbox if you want to upload only stock but no product data. 

Author: ```suggestion
23. Click the *Upload stock only* menu entry in the left side bar.  
    The upload stock only setting is displayed on the right side.

    ![Upload stock only](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_UploadStockOnly.png "[Upload stock only]")

24. Select the *Upload stock only* checkbox if you want to upload only stock but no product data. Leave the checkbox unchecked to upload both stock and product data. By default, the checkbox is unchecked.  
```

25. Click the *Order search query* menu entry in the left side bar.  

>     The order search query setting is displayed in the right side bar.
> 
>     ![Order search query](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_OrderSearchQuery.png "[Order search query]")
> 
> 26. If desired, enter a query to find out which orders have already been imported into the *Actindo Core1 Platform*. Alternatively, you can enable the *Apply from default* toggle to apply the predefined values.
> 
>     > [Info] As soon as an order has been imported, the order *Export status* is set to **Exported** in the Salesforce Commerce Cloud. The order *Export status* can be check in *Merchant Tools > Ordering > Orders > Search for an order > Select an order*. 

Author: ```suggestion
    The order search query setting is displayed on the right side.

    ![Order search query](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_OrderSearchQuery.png "[Order search query]")

26. Enter a custom query to search for all new imported orders.    
Alternatively, enable the *Apply from default* toggle to apply the default search query.

    > [Info] As soon as an order has been imported, the order *Export status* is set to **Exported** in the Salesforce Commerce Cloud. The order *Export status* can be check in *Merchant Tools > Ordering > Orders > Search for an order > Select an order*. 
```

27. Click the *Number of parallel jobs* menu entry in the left side bar.  

>     The number of parallel jobs setting is displayed in the right side bar.
> 
>     ![Number of parallel jobs](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_NumberParallelJobs.png "[Number of parallel jobs]")
> 
> 28. Enter a higher number of jobs in the *Number of parallel jobs* if the import volume capacity needs to be increased. The number of parallel jobs is directly related to the number of available VCores.  Alternatively, you can enable the *Apply from default* toggle to apply the predefined values.

Author: ```suggestion
    The number of parallel jobs setting is displayed on the right side.

    ![Number of parallel jobs](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_NumberParallelJobs.png "[Number of parallel jobs]")

28. Enter the number of jobs that may be processed in parallel in the *Number of parallel jobs* field.   
Alternatively, enable the *Apply from default* toggle to apply the default value for the number of parallel jobs.

> [Info] Increase the number of parallel jobs to handle a higher order volume. Be aware that a higher number of parallel jobs also requires a higher number of vCores. 
```

29. Click the *Custom inventory fields* menu entry in the left side bar.  
    The custom inventory fields setting is displayed in the right side bar.

    ![Custom inventory fields](../../Assets/Screenshots/Channels/Settings/Connections/SFCC/Settings_CustomInventoryFields.png "[Custom inventory fields]")

30. If desired, select the *Handle inventory custom fields* checkbox if inventory custom fields are imported and must be maintained in the system.

31. Click the [Save] button.  
    All changes have been saved. The *Saving successful* pop-up window is displayed.

    ![Saving successful](../../Assets/Screenshots/Channels/Settings/Connections/SavingSuccessful.png "[Saving successful]")

> ## Configure the Shopify ETL mapping

Author: das folgende muss komplett umgestellt werden - die meisten Punkte betreffen nicht das ETL-Mapping, sondern andere Besonderheiten der Verbindung. Dafür würde ich einen gesonderten Header machen und dann darunter (eventuell als sub-header) die einzelnen Besonderheiten beschreiben.
